lib/active_model/serializer.rb

```diff
@@ -164,6 +164,14 @@ def json_key
       end
     end
 
+    def id
+      object.id if object
+    end
+
+    def type
+      object.class.to_s.demodulize.underscore.pluralize
```

Comment:
Oh, I see what you're doing. You are:

1. relying on the serialized object's type, not the serializer name.
2. leaving it here so people can override the inferred type.

Is that it? I'm onboard with that.